src/settings.cpp

```diff
@@ -1251,45 +1251,59 @@ void Settings::LoadGrenadeInfo(std::string path)
 	Json::Value array = data[XORSTR("smokeinfos")];
 	Settings::GrenadeHelper::grenadeInfos = {};
 
-	for(Json::Value::iterator it = array.begin(); it!=array.end(); ++it)
+	for (Json::Value::iterator it = array.begin(); it != array.end(); ++it)
 	{
-		Json::Value act = *it;
+		Json::Value& act = *it;
+
 		const char* name = act[XORSTR("name")].asCString();
+
 		GrenadeType gType = (GrenadeType)act[XORSTR("gType")].asInt();
 		ThrowType tType = (ThrowType)act[XORSTR("tType")].asInt();
 		Json::Value pos = act[XORSTR("pos")];
-		Vector posVec = Vector(pos[XORSTR("x")].asFloat(), pos[XORSTR("y")].asFloat(), pos[XORSTR("z")].asFloat());
+		Vector posVec(pos[XORSTR("x")].asFloat(), pos[XORSTR("y")].asFloat(), pos[XORSTR("z")].asFloat());
 		Json::Value angle = act[XORSTR("angle")];
-		QAngle vAngle = QAngle(angle[XORSTR("x")].asFloat(), angle[XORSTR("y")].asFloat(), 0);
-		Settings::GrenadeHelper::grenadeInfos.push_back(GrenadeInfo(gType, posVec, vAngle, tType, std::string(name)));
+		QAngle vAngle(angle[XORSTR("x")].asFloat(), angle[XORSTR("y")].asFloat(), 0.f);
+		Settings::GrenadeHelper::grenadeInfos.emplace_back(gType, posVec, vAngle, tType, name);
 	}
 }
 
 void remove_directory(const char* path)
 {
-	DIR* dir;
-	dirent* pdir;
-
-	dir = opendir(path);
-
-	while ((pdir = readdir(dir)))
-	{
+    /*
+	electric26@Desktop:~/Storage/PProjects/C++$ ./a.out
```

Comment:
kinda strange

Reply:
That's the amount of time it took to do the string operations listed `ITERATIONS` times, so it relates to the use of any of those functions/operators.

Reply:
Forgot to add: I removed it